src/differentiation.jl

```diff
@@ -1,30 +1,87 @@
 ################################################################################
-#                     Jacobian and Differential Elements
+#                          DifferentiationMethods
 ################################################################################
 
 """
-    jacobian(geometry, ts; ε=1e-6)
+    DifferentiationMethod
 
-Calculate the Jacobian of a geometry at some parametric point `ts` using a simple
-finite-difference approximation with step size `ε`.
+A category of types used to specify the desired method for calculating derivatives.
+Derivatives are used to form Jacobian matrices when calculating the differential
+element size throughout the integration region.
+"""
+abstract type DifferentiationMethod end
+
+"""
+    FiniteDifference(ε=1e-6)
+
+Use to specify use of a finite-difference approximation method with a step size
+of `ε` for calculating derivatives.
+"""
+struct FiniteDifference{T <: AbstractFloat} <: DifferentiationMethod
+    ε::T
+end
+
+# If ε not specified, default to 1e-6
+FiniteDifference() = FiniteDifference(1e-6)
+
+"""
+    Analytical()
+
+Use to specify use of analytically-derived solutions for calculating derivatives.
+These solutions are currently defined only for a subset of geometry types.
+
+# Supported Geometries:
+- `BezierCurve`
+- `Line`
+- `Plane`
+- `Ray`
+- `Tetrahedron`
+- `Triangle`
+"""
+struct Analytical <: DifferentiationMethod end
+
+# Future Support:
+#   struct AutoEnzyme <: DifferentiationMethod end
+#   struct AutoZygote <: DifferentiationMethod end
+
+################################################################################
+#                                  Jacobian
+################################################################################
+
+"""
+    jacobian(geometry, ts[, diff_method])
+
+Calculate the Jacobian of a `geometry`'s parametric function at some point `ts`.
+Optionally, direct the use of a particular `differentiation method`; by default
+use analytic solutions where possible and finite difference approximations
+otherwise.
 
 # Arguments
 - `geometry`: some `Meshes.Geometry` of N parametric dimensions
 - `ts`: a parametric point specified as a vector or tuple of length N
-- `ε`: step size to use for the finite-difference approximation
+- `diff_method`: the desired `DifferentiationMethod` to use
 """
+function jacobian end
+
+function jacobian(
+        geometry::G,
+        ts::V
+) where {G <: Geometry, V <: Union{AbstractVector, Tuple}}
+    return jacobian(geometry, ts, _default_method(G))
+end
+
 function jacobian(
         geometry::Geometry,
-        ts::V;
-        ε = 1e-6
+        ts::V,
+        diff_method::FiniteDifference
```

Comment:
Why not specifying the default here `diff_method::FiniteDifference = _default_method(Geometry)` and delete the method above?

Reply:
My goal here was to avoid conditional blocks by leveraging dispatch to define three-arg methods specific to each DM type. If the default DM were always `FiniteDifference` then I think this note would be a good idea, but the default could be any DM type, which could require dispatch to a different method than this one. In that sense, I think it makes the code easier to reason about (and more flexible moving forward) when the sole two-argument method is simply a wrapper over the three-argument with an inferred `diff_method` argument.

Reply:
Makes sense.